native-authentication/src/test/java/com/okta/nativeauthentication/form/FormFactoryTest.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2022-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.nativeauthentication.form
+
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.runBlocking
+import org.junit.Test
+
+internal class FormFactoryTest {
+    @Test fun testFormFactoryEmitsForm() = runBlocking {
+        val channel = Channel<Form>(capacity = Channel.BUFFERED)
+        val formFactory = FormFactory(channel, emptyList())
+        val formBuilder = Form.Builder()
+        val labelBuilder = Element.Label.Builder(null)
+        labelBuilder.text = "Fake Label"
+        formBuilder.elements += labelBuilder
+        formFactory.emit(formBuilder)
+        val form = channel.receive()
+        assertThat(form.elements).hasSize(1)
+        assertThat((form.elements[0] as Element.Label).text).isEqualTo("Fake Label")
+    }
+
+    @Test fun testFormFactoryEmitsMultipleForms() = runBlocking {
+        val channel = Channel<Form>(capacity = Channel.BUFFERED)
+        val formFactory = FormFactory(channel, emptyList())
+
+        for (i in 1..2) {
+            val formBuilder = Form.Builder()
+            val labelBuilder = Element.Label.Builder(null)
+            labelBuilder.text = "Fake Label $i"
+            formBuilder.elements += labelBuilder
+            formFactory.emit(formBuilder)
+        }
+
+        val form1 = channel.receive()
+        assertThat(form1.elements).hasSize(1)
+        assertThat((form1.elements[0] as Element.Label).text).isEqualTo("Fake Label 1")
+
+        val form2 = channel.receive()
```

Comment:
Is it possible to check channel capacity at the end of the test?

Reply:
Not without implementing the interface yourself. Often what you'd do is close the channel, and await the close event. Which is what something like https://github.com/cashapp/turbine would be useful for.

For this particular case, I can't think of a reasonable way to do it.